### ISSUE TITLE
`sync --remove-unused` should respect `push.namespaces` 

### ISSUE THREAD

**Issue by gu-stav** [tolgee/tolgee-cli#137]
When running `sync --remove-unused` I believe there currently is no way to configure from which namespaces keys are removed. In our case we have an API (with an `api` NS) and a CSR app (NS `app`) and syncing either one of those apps leads to the loss of keys from a NS that is managed by that app.

I think `--remove-unused` either needs a way to pass in which namespaces it affects, or it should respect the `namesspaces` setting in `push.namespaces` which would imo be preferred.

`.tolgeerc`

```json
{
  "defaultNamespace": "app",
  "patterns": ["./src/**/*.ts?(x)", "./src/**/*.vue"],
  "push": {
    "forceMode": "OVERRIDE",
    "namespaces": ["app", "shared"]
  }
}
```

I think the CLI has all information at hand already, because `compare` does not only list messages which are up for deletion, but also their namespace.

**Comment by stepan662** [comment]
Yes you are right, this is option is not currently available. However it was not intended to work this way and I'm not a fan of sharing of these options. The `push` and `sync` commands are quite a bit different.

I'm tagging this as `enhancement`, we can add this feature later.

**Comment by gu-stav** [comment]
I understand and thought so tbh. Thanks for the quick response.

Would you be willing to accept a PR for that? I could either explore the route using a CLI argument (which would probably the easiest) or introducing a new `sync` attribute on the config?

The latter could work similar to `push` I suppose:

```js
{
  "sync": {
    "namespaces": ["app", "shared"]
  }
}
```

Can't promise anything but if you are open to a PR I would look into it to get a feeling how much time I'd have to spent ✌🏼

**Comment by stepan662** [comment]
PR would be cool 🙂 I'm in favor having the option separately. The should be basically mapping the CLI options logic, so when it's a separate command, I would put it to separate section.
